js/pages/cohort-definitions/cohort-definition-manager.js

```diff
@@ -231,11 +231,11 @@ define(['jquery', 'knockout', 'text!./cohort-definition-manager.html',
 			this.isFeMaleSample=ko.observable(false);
 			this.isOtherGenderSample=ko.observable(false);
 			//error state
-			this.isAgeRange =ko.pureComputed(() => ['between','notBetween'].includes(this.sampleAgeType()));
-			this.isAgeRangeError = ko.pureComputed(() => this.isAgeRange() && (this.firstAge() == null || this.secondAge() == null));
-			this.firstAgeError = ko.pureComputed(() => this.firstAge() == null || this.firstAge() < 0);
 			this.sampleNameError=ko.pureComputed(() => this.sampleName().trim() == "");
 			this.patientCountError=ko.pureComputed(() => !(this.patientCount() > 0)); // this works because null == 0
+			this.isAgeRange =ko.pureComputed(() => ['between','notBetween'].includes(this.sampleAgeType()));
+			this.firstAgeError = ko.pureComputed(() => this.firstAge() != null && this.firstAge() < 0);
+			this.isAgeRangeError = ko.pureComputed(() => this.isAgeRange() && (this.firstAgeError() || (this.secondAge() != null && this.secondAge() < 0) || this.firstAge() == this.secondAge()));
```

Comment:
Not sure if this is encoded, it think that another age range condition is `firstAge <= secondAge`.

Reply:
Actually, the age order doesn't matter (I was surprised to find this) that's why the validation check says that it's an error if they are equal.  When the form is submitted, the smaller value is put as firstAge, and the larger is secondAge.